core/src/banking_stage/latest_unprocessed_votes.rs

```diff
@@ -274,49 +274,42 @@ impl LatestUnprocessedVotes {
         bank: Arc<Bank>,
         forward_packet_batches_by_accounts: &mut ForwardPacketBatchesByAccounts,
     ) -> usize {
-        let mut continue_forwarding = true;
         let pubkeys_by_stake = weighted_random_order_by_stake(
             &bank,
             self.latest_votes_per_pubkey.read().unwrap().keys(),
         )
         .collect_vec();
-        pubkeys_by_stake
-            .into_iter()
-            .filter(|&pubkey| {
-                if !continue_forwarding {
-                    return false;
-                }
-                if let Some(lock) = self.get_entry(pubkey) {
-                    let mut vote = lock.write().unwrap();
-                    if !vote.is_vote_taken() && !vote.is_forwarded() {
-                        let deserialized_vote_packet = vote.vote.as_ref().unwrap().clone();
-                        if let Some(sanitized_vote_transaction) = deserialized_vote_packet
-                            .build_sanitized_transaction(
-                                bank.vote_only_bank(),
-                                bank.as_ref(),
-                                bank.get_reserved_account_keys(),
-                            )
-                        {
-                            if forward_packet_batches_by_accounts.try_add_packet(
-                                &sanitized_vote_transaction,
-                                deserialized_vote_packet,
-                                &bank.feature_set,
-                            ) {
-                                vote.forwarded = true;
-                            } else {
-                                // To match behavior of regular transactions we stop
-                                // forwarding votes as soon as one fails
-                                continue_forwarding = false;
-                            }
-                            return true;
+        let mut count: usize = 0;
+        for pubkey in pubkeys_by_stake {
+            if let Some(lock) = self.get_entry(pubkey) {
+                let mut vote = lock.write().unwrap();
+                if !vote.is_vote_taken() && !vote.is_forwarded() {
+                    let deserialized_vote_packet = vote.vote.as_ref().unwrap().clone();
+                    if let Some(sanitized_vote_transaction) = deserialized_vote_packet
+                        .build_sanitized_transaction(
+                            bank.vote_only_bank(),
+                            bank.as_ref(),
+                            bank.get_reserved_account_keys(),
+                        )
+                    {
+                        if forward_packet_batches_by_accounts.try_add_packet(
+                            &sanitized_vote_transaction,
+                            deserialized_vote_packet,
+                            &bank.feature_set,
+                        ) {
+                            vote.forwarded = true;
```

Comment:
Using the "early return" pattern, the indentation for this whole for loop body can be reduced by a lot.
I think it is easier to follow this way:

```suggestion
        for pubkey in pubkeys_by_stake {
            let Some(vote) = self.get_entry(pubkey) else {
                continue;
            };
            let mut vote = vote.write().unwrap();

            if !vote.is_vote_taken() && !vote.is_forwarded() {
                continue;
            }

            let deserialized_vote_packet = vote.vote.as_ref().unwrap().clone();

            let Some(sanitized_vote_transaction) =
                deserialized_vote_packet.build_sanitized_transaction(
                        &bank.feature_set,
                        bank.vote_only_bank(),
                        bank.as_ref(),
                    )
            else {
                continue;
            };

            if !forward_packet_batches_by_accounts.try_add_packet(
                &sanitized_vote_transaction,
                deserialized_vote_packet,
                &bank.feature_set,
            ) {
                // To match behavior of regular transactions we stop forwarding votes as soon as one
                // fails. We are assuming that failure (try_add_packet) means no more space
                // available.
                break;
            }

            vote.forwarded = true;
            count += 1;
        }
```

Reply:
Good catch.

Hopefully, there is a unit test that would have failed :)

P.S. Adding some paragraphs would not hurt, I think.
Writing everything with no vertical space makes it look like it is a single combined statement to me %)